core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -813,41 +813,56 @@ class Partition(val topicPartition: TopicPartition,
    */
   private def tryCompleteDelayedRequests(): Unit = delayedOperations.checkAndCompleteAll()
 
+  // Visible for testing.
+  private[cluster] def shouldShrinkIsr(): Boolean = {
+    leaderLogIfLocal match {
+      case Some(leaderLog) =>
+        inReadLock(leaderIsrUpdateLock) {
```

Comment:
It's probably not doing much harm, but I think we don't need the read lock here. The fields we access seem safe for concurrent access and we redo the check anyway in `maybeShrinkIsr`.

Reply:
You're right, that appears to be the case. Done.